### DISCUSSION TITLE
Any way to launch a non steam game??

### DISCUSSION THREAD

**Discussion by totalitarian** [ebkr/r2modmanPlus#969] [Q&A]
If my version of a game that I want to mod is installed in steam by the 'non steam game' button, is there still a way to use r2modman?

**Comment by ebkr** [comment]
We don’t support pirated copies

**Reply by j-loewen** [reply]
What about Microsoft Store games? I have installed Cities Skyline 2 (Free with Game Pass) through the Microsoft Store. Can I use the Mod Manager for this? And if yes, how?

**Reply by arp-mbender** [reply]
I recently got 20 Minutes Till Dawn from EGS. I'd like to mod it all the same. There'll also be plenty of games from GOG.

Do NOT assume non-Steam games are immediately pirated - that's just rude.

**Reply by Bread1009** [reply]
i agree! i have several games from gog id like to mod as well. i assume there is no way to use r2mm to do this?

**Reply by ebkr** [reply]
> Do NOT assume non-Steam games are immediately pirated - that's just rude.

This is usually the case for when people ask. It's not rude to assume that's the case given there was no other context.

As for games available on other stores, those can be supported, I just need to know the game identifiers.
For example, Dyson Sphere Program's identifier on the Microsoft Store is: `GameraGame.DysonSphereProgram`. I also need to know the name of the game folder and the executable.

For games on GoG, they won't have a store identifier, I just need to know the executable and game folder name

**Reply by Klerdante2137** [reply]
well i have pirated copy of Ultrakill with how Hakita said "Culture can't exist for only those who can afford it" and said if you can't buy game spread about it via word of mouth. coming to that conclusion even if someone has pirated copy there should be way to boot mods from thunderstore. i tried setting game folder to new non steam direction but it dosen't work

**Reply by hitmangmod** [reply]
you guys are not helping if you cant help then dont or please do not give misinformation 
heres how to do it for basically any game: go to the launcher then settings go to debugging find the parameters copy the moded parameters go to the game location make a shortcut of said game go to properties click on the target at the end right after " game location" put a space and add --doorstop-enable true --doorstop-target "C:\Users\youruser\AppData\Roaming\r2modmanPlus-local\Valheim\profiles\salam\BepInEx\core\BepInEx.Preloader.dll"
edited the swear words but if my guide is too short or complicated user [StuckProcessing] has provided full step by step down below https://github.com/ebkr/r2modmanPlus/discussions/969#discussioncomment-12904101

**Reply by XashAEXII** [reply]
> you guys are fcing useless if you cant help then dont or stf up heres how to do it for basically any game: go to the launcher then settigns go to debugging fidn the parameters copy the moded parameters go to the game location make a shortcut of said game go to properties click on the traget at the end right after " game lcoation" put a space and add --doorstop-enable true --doorstop-target "C:\Users\youruser\AppData\Roaming\r2modmanPlus-local\Valheim\profiles\salam\BepInEx\core\BepInEx.Preloader.dll"

i do not understand. can you be more especific ? [maybe is just me but, plz help]

**Reply by StuckProcessing** [reply]
Settings > Debugging tab > Set Launch Parameters

Copy the "Modded:" parameters, they will look like this ("username" will be your Windows profile name, "profilename" will be the name of your profile):
`--doorstop-enable true --doorstop-target "C:\Users\username\AppData\Roaming\r2modmanPlus-local\ULTRAKILL\profiles\profilename\BepInEx\core\BepInEx.Preloader.dll"`

Create a shortcut to the game's executable (for this example it is ULTRAKILL.exe), and in the "Target" box, append the "Modded" parameters to the rest of the text. It will look like this (Obviously this may not be the path you have installed the game. Also, again, you need to change "username" to your Windows profile name, and profilename to the name of your profile):
`"C:\Games\Regular downloading\Legally obtained\ULTRAKILL\ULTRAKILL.exe" --doorstop-enable true --doorstop-target "C:\Users\username\AppData\Roaming\r2modmanPlus-local\ULTRAKILL\profiles\profilename\BepInEx\core\BepInEx.Preloader.dll"`

You will need to repeat these steps if you want to use a different Profile

**Reply by FrozenBogan** [reply]
> you guys are fcing useless if you cant help then dont or stf up heres how to do it for basically any game: go to the launcher then settigns go to debugging fidn the parameters copy the moded parameters go to the game location make a shortcut of said game go to properties click on the traget at the end right after " game lcoation" put a space and add --doorstop-enable true --doorstop-target "C:\Users\youruser\AppData\Roaming\r2modmanPlus-local\Valheim\profiles\salam\BepInEx\core\BepInEx.Preloader.dll"

Any idea how to get around "unable to resolve steam install file"

I don't have steam on the computer but R2 wants a steam.exe location

**Reply by StuckProcessing** [reply]
> Any idea how to get around "unable to resolve steam install file"
> 
> I don't have steam on the computer but R2 wants a steam.exe location

Don't launch the game through r2modman. Run the game through the game's shortcut

**Reply by FrozenBogan** [reply]
> Don't launch the game through r2modman. Run the game through the game's shortcut

Haha thanks mate, works a treat! Cheers for the patience and assistance

**Reply by XashAEXII** [reply]
> Settings > Debugging tab > Set Launch Parameters

 >the setting in what ?

nevermind, i noticed already[the r2modman settigs you mean]. thanks for the help

**Reply by XashAEXII** [reply]
>i do need to have steam instaled ?
>it needs to have an account ?

no,neither of both

**Reply by dgeauyetyuregiuoghihejhdakjhdjahi** [reply]
_**then support them idiot**_

**Reply by breadbutcooler** [reply]
> We don’t support pirated copies

so using gog or any other non steam game market is piracy apparently?

**Reply by BouncyBonBon** [reply]
> Create a shortcut to the game's executable (for this example it is ULTRAKILL.exe), and in the "Target" box, append the "Modded" parameters to the rest of the text. It will look like this (Obviously this may not be the path you have installed the game. Also, again, you need to change "username" to your Windows profile name, and profilename to the name of your profile): `"C:\Games\Regular downloading\Legally obtained\ULTRAKILL\ULTRAKILL.exe" --doorstop-enable true --doorstop-target "C:\Users\username\AppData\Roaming\r2modmanPlus-local\ULTRAKILL\profiles\profilename\BepInEx\core\BepInEx.Preloader.dll"`

Looks like this has been patched unfortunately, using either these launch parameters or edited versions of those in the launch params menu still forces a boot off of the Steam build I have (even starting a download from Steam if the game is uninstalled). I'm trying to run off of an old build which I downloaded off of Steam's console (I'm using a mod that only runs on a specific patch of the game).

**Reply by probablykate** [reply]
mf thinks non steam = pirated, have you ever heard of gog.com?

**Reply by L-Wys** [reply]
What about Xbox Gamepass? Doesn't seem to work with that and those are clearly not pirated copies??? 
Piracy doesn't just equal bad, mate, and making it rather inaccesible in the name of good/not supporting piracy is just infuriatingly stupid.

I just want to play an archipelago and don't have the money spare to buy a full price game that I just have on Gamepass.

**Reply by Rings4** [reply]
Surely a developer isn't this conceited to think that Steam is the ONLY available platform for PC games??

**Comment by dgeauyetyuregiuoghihejhdakjhdjahi** [comment]


**Reply by soulware1** [reply]
epic game store users in shambles

**Reply by dgeauyetyuregiuoghihejhdakjhdjahi** [reply]
wait this is my account? i didn't post that

**Comment by Zytyhd** [comment]
Tried all solutions up and nothing worked i still get no license thing.

**Reply by soulware1** [reply]
i doubt you actually did that,
make a shortcut of the executable, go into its properties, and add --doorstop-enable true --doorstop-target "%appdata%\r2modmanPlus-local\Valheim\profiles\salam\BepInEx\core\BepInEx.Preloader.dll"
(Note: also in the app itself it tells you how to do this, so go there and into help if this doesn't work)

**Comment by ccccool34gf** [comment]
I also would like to know how to launch non-steam games. I bought through GOG and can't launch the game because it wants a steam code

**Reply by StuckProcessing** [reply]
https://github.com/ebkr/r2modmanPlus/discussions/969#discussioncomment-12904101

**Comment by dgeauyetyuregiuoghihejhdakjhdjahi** [comment]
**previous comment deleted: it was on my account and i don't remember posting that. pretty sure i didn't.**
i wish. i think a workaround would be using 'add non-steam game' or whatever it was.
![crimsonboingboing](https://github.com/user-attachments/assets/7af7c613-24ab-46a4-a7ec-10f10eaefa40)
